### PR TITLE
Increse test coverage for the relay

### PR DESCRIPTION
**Pull request by ensonic**
* Helper Functions: Added unit tests for marshalHeader, unmarshalHeader, and extractBackendNameAndPath, including edge cases and error conditions.
* Server Handlers:
  * Added tests for the health handler.
  * Improved error path coverage for serverRequestStream, serverResponse, and serverRequest (e.g., missing parameters, invalid proto bodies).
  * Added TestNewServerConfig to verify default and custom configurations.
  * Added TestResponseFilterChunking to cover asynchronous response streaming.
  * Added TestResponseTrailers and TestGrpcHeadersAsTrailers to verify trailer handling and conversion of "Grpc-" headers.
* Broker:
  * Added tests for RelayRequest error conditions (URL parsing errors, multiple clients for the same request ID).
  * Added TestGetRequestServerRestarting to cover server shutdown scenarios.
  * Ensured that inactivity timeouts and relay timeouts are handled correctly (though some tests are skipped in Short mode to avoid long delays).

### CODE REVIEW

File: src/go/cmd/http-relay-server/server/broker_test.go [hunk 1]
```diff
@@ -351,3 +351,75 @@ func TestReapWhileSendingRequest(t *testing.T) {
...
+	}
+}
+
+func TestGetRequestServerRestarting(t *testing.T) {
+	b := newBroker()
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel() // Cancel immediately
+
+	_, err := b.GetRequest(ctx, "foo", "/")
+	if !errors.Is(err, ErrServerRestarting) {
```
Comment (csieber) [inline src/go/cmd/http-relay-server/server/broker_test.go:389]: `ErrServerRestarting` is not intuitive here. Can we add a comment why the cancelled context on the request results in this error?

File: src/go/cmd/http-relay-server/server/broker_test.go [hunk 2]
```diff
@@ -351,3 +351,75 @@ func TestReapWhileSendingRequest(t *testing.T) {
...
+		t.Errorf("Expected ErrRelayTimeout, got %v", err)
+	}
+}
+
+func TestGetRequestStreamTimeout(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping slow timeout test")
+	}
+	b := newBroker()
+	b.req["foo"] = make(chan *pb.HttpRequest, 1)
```
Comment (csieber) [inline src/go/cmd/http-relay-server/server/broker_test.go]: Also add a comment there that the channel blocks indefinitely.

File: src/go/cmd/http-relay-server/server/server_test.go [hunk 3]
```diff
@@ -186,6 +188,101 @@ func TestClientHandlerWithChunkedResponse(t *testing.T) {
...
+
+	marshaled := marshalHeader(&h)
+	if len(marshaled) != 3 {
+		t.Errorf("Expected 3 marshaled headers, got %d", len(marshaled))
+	}
+
+	w := httptest.NewRecorder()
+	unmarshalHeader(w, marshaled)
+
+	if got := w.Header().Values("X-Test"); len(got) != 2 || got[0] != "value1" || got[1] != "value2" {
```
Comment (csieber) [inline src/go/cmd/http-relay-server/server/server_test.go:206]: Does this assume the header order is stable? Is this true?

File: src/go/cmd/http-relay-server/server/server_test.go [hunk 4]
```diff
@@ -427,3 +524,185 @@ func BenchmarkCreateId(b *testing.B) {
...
+	})
+
+	wg.Wait()
+	resp := respRecorder.Result()
+	if got := resp.Trailer.Get("X-Trailer"); got != "value" {
+		t.Errorf("Expected trailer X-Trailer: value, got %q", got)
+	}
+}
+
+func TestGrpcHeadersAsTrailers(t *testing.T) {
```
Comment (csieber) [inline src/go/cmd/http-relay-server/server/server_test.go]: This test case is very similar to the previous one. Do we really need both?

**Review by csieber — APPROVED** [review]

File: src/go/cmd/http-relay-server/server/broker_test.go (hunk 1 above)
Comment (ensonic) [reply src/go/cmd/http-relay-server/server/broker_test.go:389]: I think I should rename the error. There is nothing restarting.
The error is returned to serverRequest()
https://github.com/googlecloudrobotics/core/blob/main/src/go/cmd/http-relay-server/server/server.go#L481

Okay if I do the rename in a followup?

File: src/go/cmd/http-relay-server/server/broker_test.go (hunk 2 above)
Comment (ensonic) [reply src/go/cmd/http-relay-server/server/broker_test.go]: Done. We shoudl consider to also add a context where.

File: src/go/cmd/http-relay-server/server/broker_test.go (hunk 1 above)
Comment (csieber) [reply src/go/cmd/http-relay-server/server/broker_test.go:389]: SGTM

File: src/go/cmd/http-relay-server/server/server_test.go (hunk 3 above)
Comment (ensonic) [reply src/go/cmd/http-relay-server/server/server_test.go:206]: I think it is (it is a slice). Also this tests the marshalling that preserves the order.

File: src/go/cmd/http-relay-server/server/server_test.go (hunk 4 above)
Comment (ensonic) [reply src/go/cmd/http-relay-server/server/server_test.go]: merged both into a table driven test. Better?

Comment (csieber) [reply src/go/cmd/http-relay-server/server/server_test.go]: looks good!
